deploy/03_deploy_vending_machine.ts

```diff
@@ -0,0 +1,39 @@
+import { HardhatRuntimeEnvironment } from "hardhat/types"
+import { DeployFunction } from "hardhat-deploy/types"
+
+const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
+  const { getNamedAccounts, deployments } = hre
+  const { deployer } = await getNamedAccounts()
+  const { read } = deployments
+
+  const KeepToken = await deployments.get("KeepToken")
+  const T = await deployments.get("T")
+
+  const keepTotalSupply = await read("KeepToken", "totalSupply")
+  const tTotalSupply = await read("T", "totalSupply")
+
+  // We're wrapping 100% of the minted KEEP and will be allocating 45% of the
+  // minted T tokens. The remaining T tokens will be in the future distributed
+  // between another instance of the VendingMachine (which will be wrapping NU
+  // token) and a DAO treasury.
+
+  const T_ALLOCATION = tTotalSupply.mul(45).div(100)
+
+  const vendingMachine = await deployments.deploy("VendingMachine", {
+    from: deployer,
+    args: [KeepToken.address, T.address, keepTotalSupply, T_ALLOCATION],
+    log: true,
+  })
```

Comment:
We will be deploying two Vending Machines: one for KEEP and one for NU. Let's declare the one for Keep:

```suggestion
const vendingMachineKeep = await deployments.deploy("VendingMachineKeep", {
    contract: "VendingMachine",
    from: deployer,
    args: [KeepToken.address, T.address, keepTotalSupply, T_ALLOCATION],
    log: true,
  })
```

Reply:
Changed, see f142285fc67da3afa607c8fa8ff58d98b37416ae.